Add docs to clustering template notebook (#210)
AlexsLemonade/scpca-downstream-analyses: optional-clustering-analysis/clustering-report-template.Rmd

    code_folding: hide
---

> This notebook will plot the output of the 
> `optional-clustering-analysis/clustering-calculations.R` script, which allows 
> for additional clustering parameters to be calculated on gene expression data 
> associated with a single sample of the dataset of interest.
> 

Reviewer: ```suggestion
This notebook will plot the output of the 
`optional-clustering-analysis/clustering-calculations.R` script, which allows 
for additional types of clustering across a range of parameters to be calculated on gene expression data 
associated with a single sample of the dataset of interest.

```
I just want to make sure we specify that it could be both additional types of clustering and a range of parameters.

## UMAP plot results for `r params$library`

> Below is a UMAP plot for each clustering parameter, where each cell has been 

Reviewer: ```suggestion
Below is a set of UMAP plots, where each panel corresponds to clustering performed with a given clustering parameter and each cell is
```

cowplot::plot_grid(plotlist = plot_list, ncol = num_col)
```

## Evaluating clustering

Below we evaluate the clustering results across each of the parameters tested. 
We will look at the cluster purity, silhouette width, and cluster stability for 
each clustering type and nearest neighbors value used.
For a more in depth discussion on these metrics and how they can be used to 
identify the optimal clustering results, see the [advanced clustering chapter in Orchestrating Single Cell Analysis](http://bioconductor.org/books/3.15/OSCA.advanced/clustering-redux.html#motivation).

#### Purity plots

The purity plots below represent each cell as the proportion of neighboring 
cells that are assigned to the same cluster.
Well-separated clusters should show little overlap between member and 
neighboring cells, and therefore high purity values for all member cells.

> The cluster purity range is `0` to `1`. Therefore purity values that are

Reviewer 2: OSPL; this may be my fault for a previous suggestion!

> ## Plot cluster validity stats
> 
> Below are plots that display the validity results of each of the cluster assignments.

Reviewer: ```suggestion
## Evaluating clustering

Below we evaluate the clustering results across each of the parameters tested. 
We will look at the cluster purity, silhouette width, and cluster stability for each clustering type and nearest neighbors value used.
For a more in depth discussion on these metrics and how they can be used to identify the optimal clustering results, see the [advanced clustering chapter in Orchestrating Single Cell Analysis](http://bioconductor.org/books/3.15/OSCA.advanced/clustering-redux.html#motivation).
```
I would focus on what these plots are showing us rather than just stating we are making plots. The purpose of these plots is to evaluate the clustering results for each of the parameters tested and be able to compare those outputs to each other in a quantitative fashion.

#### Purity plots

The purity plots below represent each cell as the proportion of neighboring 
cells that are assigned to the same cluster.

> Well-separated clusters should show little over-lapping between member and 

Reviewer 2: ```suggestion
Well-separated clusters should show little overlap between member and 
```

neighboring cells, and therefore high purity values for all member cells.

> The cluster purity range is `0` to `1`, therefore purity values that are

Reviewer 2: This should actually be two sentences - 

```suggestion
The cluster purity range is `0` to `1`. Therefore purity values that are
```

### Summary plots

> The summary plots below represent a calculated average for each type of cluster 
> validity statistic, cluster purity and silhouette width, across each of the 
> cluster parameters.

Reviewer: ```suggestion
The summary plots below represent a calculated average for each type of the clustering metrics, cluster purity and silhouette width, across each of the parameters tested.
```

> Again, ideally you would expect high cluster purity values and large positive

Reviewer 2: ```suggestion
Again, robust clusters should yield high cluster purity values and large positive
```

## Plot cluster stability

The plots below represent how stable the clustering results associated with each

> of the clustering parameters appear to be when using bootstrap replicates.

Reviewer 2: ```suggestion
of the clustering parameters appear to be, as assessed using bootstrap replicates.
```

> The plots below represent how stable the clustering results associated with each
> of the clustering parameters appear to be when using bootstrap replicates.

Reviewer: ```suggestion
The plots below represent the stability of the clustering results associated with each
of the clustering parameters.
Here, cells within each dataset are sampled using bootstrapping and the sampled cells are re-clustered. 
The new clustering assignments are compared to the original assignment by obtaining an adjusted rand index, and this process is repeated 20 times.
```
I would expand a little on the methods here since this is a complex process and we used our own methods rather than an available R package.

> Ideally, bootstrap replicates would be highly consistent with the original 
> dataset, as this would mean that clusters are likely to be stable despite small 
> changes to upstream processing and overall would mean that they can be reproduced.

Reviewer: ```suggestion
Clustering results with high stability would reveal that clustering after each bootstrap replicate is consistent with the original clustering results.
```
I would watch out that we don't overstate the conclusions. Even if a clustering result is stable, that doesn't necessarily mean that an upstream change will not change the clustering results.

We use summary adjusted Rand Index (ARI) values in the plots below to represent 
the average calculated cluster stability values.

> The range here is `0` to `1`, where clusters would ideally have ARI values that

Reviewer 2: Rephrase -
```suggestion
The range here is `0` to `1`, where stable clusters have values closer to 1.
```

> are closer to 1, as this would mean greater cluster stability.

Reviewer 2: Delete this if you use my last suggestion
```suggestion
```
